consensus/bor/bor.go

```diff
@@ -455,14 +449,24 @@ func (c *Bor) snapshot(chain consensus.ChainHeaderReader, number uint64, hash co
 				// get checkpoint data
 				hash := checkpoint.Hash()
 
+				state, err := chain.StateAt(hash)
```

Comment:
Getting state `nil` here in case of 0 block number.
@jdkanani @ferranbt Please review.